src/tests/ftest/util/spdk.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+
+import getpass
+import threading
+import Queue
+
+from ssh_connection import Ssh
+
+SPDK_SETUP_SCRIPT = "/opt/daos/spdk/scripts/setup.sh"
+
+class SpdkFailed(Exception):
+    """ SPDK Setup/Cleanup did not work """
+
+def nvme_setup_thread(result_queue, hostname, debug=True):
+    """
+    nvme setup thread function
+    Arge:
+        result_queue: Queue for PASS/FAIL
+        hostname: server name
+        debug: print the debug message, default True
+    """
+    host = Ssh(hostname, debug=debug)
+    host.connect()
+    cmd = ["sudo /usr/bin/mkdir -p /opt/daos",
+           "sudo /usr/bin/git clone https://github.com/spdk/spdk.git" \
+           " /opt/daos/spdk",
```

Comment:
Is a checkout of an entire git repo necessary here?  What exactly from the repo are you using?

Reply:
But these should be available in `install/` shouldn't they?

Reply:
It is available in install. The issue is setup.sh script needs to be run as super user. Way we got this supported from DCO is freeIPA, where specific set of command can be run as sudo by any user or in automation script. FreeIPA need fix path for command so we can not use the user given path for setup.sh. We decided to use the fix path after checking out the SPDK repo to location (/opt/daos) and use it for automation purpose.

Reply:
IIRC, `sudo` configuration can contain wildcard matches in the command path, so DCO should be able to create sudo rules that match any possible run-time configuration so that the commands from the current `install/` can be used.  I have done this in previous projects.

Reply:
That is what I requested first, Ypu can find more detail in ticket [DCO-8268](https://jira.hpdd.intel.com/browse/DCO-8268?focusedCommentId=233955&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-233955 )

Reply:
NVMe setup/cleanup code has been updated based on RPM now based on comment [DCO-8268 Comment](https://jira.hpdd.intel.com/browse/DCO-8268?focusedCommentId=235672&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-235672)